src/javascript/track.js

```diff
@@ -127,11 +127,12 @@ export default class DropTrack {
 
     // note update
     for (let i = 0; i < this.noteArr.length; ++i) {
-      this.noteArr[i].update(i === 0 && this.isUserHoldingNote);
-      if (this.noteArr[i].isOutOfCanvas()) {
-        this.noteArr.splice(i, 1); // Remove out of canvas note
-        --i; // Correct the index value
-      }
+      const isUserHolding = i === 0 && this.isUserHoldingNote;
```

Comment:
Might want to document what is special about the first element in the `noteArr`. Specifically, are you using `noteArr[0]` as some sentinel value / flag?

Reply:
Oh yes... noteArr[0] means the closest note (i.e. the next note) that the user is going to dismiss, I'll add a comment now